src/js/actions/appActions.js

```diff
@@ -143,12 +143,18 @@ export const setFirstLoginAfterSignup = firstLoginAfterSignup => dispatch =>
     firstLoginAfterSignup: firstLoginAfterSignup
   });
 
-export const setYesterday = () => dispatch => {
+const dateFunctionMap = {
+  getDays: 'getDate',
+  setDays: 'setDate'
+};
+
+export const setYesterday = () => (dispatch, getState) => {
+  const { interval, intervalUnit } = getOfflineThreshold(getState());
   const today = new Date();
-  const intervalName = `${DEVICE_ONLINE_CUTOFF.intervalName.charAt(0).toUpperCase()}${DEVICE_ONLINE_CUTOFF.intervalName.substring(1)}`;
-  const setter = `set${intervalName}s`;
-  const getter = `get${intervalName}s`;
-  today[setter](today[getter]() - DEVICE_ONLINE_CUTOFF.interval);
+  const intervalName = `${intervalUnit.charAt(0).toUpperCase()}${intervalUnit.substring(1)}`;
+  const setter = dateFunctionMap[`set${intervalName}`] ?? `set${intervalName}`;
+  const getter = dateFunctionMap[`get${intervalName}`] ?? `get${intervalName}`;
+  today[setter](today[getter]() - interval);
 
   return Promise.resolve(dispatch({ type: AppConstants.SET_YESTERDAY, value: today.toISOString() }));
```

Comment:
good point in this case (since we only have minutes, hours & days for now)... in general this is more of an issue for months & years - if we would only check seconds, we will be off around February (or around leap years), in the date functions these considerations are already handled...